lib/digest/crc.rb

```diff
@@ -93,10 +96,52 @@ def digest_length
     # @param [String] data
     #   The data to update the CRC checksum with.
     #
-    # @abstract
-    #
     def update(data)
-      raise(NotImplementedError,"#{self.class}##{__method__} not implemented")
+      unless @width >= 1
+        raise(NotImplementedError, "incompleted #{self.class} as CRC (expected WIDTH to be 1 or more)")
+      end
+
+      if @table.empty?
+        raise(NotImplementedError, "incompleted #{self.class} as CRC (expected TABLE to be not empty)")
+      end
+
+      if self.class::REFLECT_INPUT.nil?
+        raise(NotImplementedError, "incompleted #{self.class} as CRC (expected REFLECT_INPUT to be not nil)")
+      end
+
+      table = @table
+      crc = @crc
+
+      if self.class::REFLECT_INPUT
+        if @width > 8
+          data.each_byte do |b|
+            crc = table[b ^ (0xff & crc)] ^ (crc >> 8)
+          end
+        else
+          data.each_byte do |b|
+            # Omit (crc >> 8) since bits upper than the lower 8 bits are always 0
+            crc = table[b ^ (0xff & crc)]
+          end
+        end
+      else
+        if @width > 8
+          drops = @width - 8
```

Comment:
Is there possibly a better variable name then `drops` that explains what it's doing to the CRC bits?

Reply:
Changed to `higher_bit_off`.